[nvfuser] avoid out of bounds error

davidberard98 opened the pull request:
Summary: update OOB check (https://github.com/csarofeen/pytorch/pull/2218) and skip tests that OOM on internal machines.

Test Plan:
```
buck2 test mode/dev-nosan //caffe2/torch/csrc/jit/codegen/cuda/test:nvfuser
```

Differential Revision: D41502369

jjsjann123 commented on torch/csrc/jit/codegen/cuda/transform_view.cpp:
```diff
@@ -510,7 +510,8 @@ class AnalyzeViewTransformation {
       if ((new_view_index == new_view_.size() ||
-           (new_view_[new_view_index + 1] != 1)) &&
+           (new_view_.size() > new_view_index + 1 &&
+            new_view_[new_view_index + 1] != 1)) &&
```
This actually looks about right to me as well. This case should be handled later at
line 566:
`if (current_size == 1 && isImplicitBroadcast(original_view_index)) {`

Any chance you have a repro of the failing case?

davidberard98 replied:
Repro: `FusionViewAllShmoo_CUDA` test; I think `new_view_[new_view_index + 1]` (above) is the out-of-bounds access. Internally we have a build mode that catches these out of bound accesses, you can probably add an assert for the access to do the same check

jjsjann123 replied:
hmmm... There's an actual cpp tests that's failing?! Let me double check that.

@naoya has been advocating us abandoning `std::vector::operator[]` :)

jjsjann123 reviewed (approved):
OOM is surprising. But let's get them in first since it's breaking tests.
I'll cherry-pick Naoya's fix internally after code bump.

davidberard98 commented:
@pytorchbot merge

davidberard98 commented:
@pytorchbot rebase -s

davidberard98 committed:
[nvfuser] patch tests for internal CI (#89584)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/89584

new_view_[new_view_index + 1] was hitting OOB errors.

Test Plan:
```
buck2 test mode/dev-nosan //caffe2/torch/csrc/jit/codegen/cuda/test:nvfuse
```

^^ TODO, this is failing with assertion errors now.

https://www.internalfb.com/intern/testinfra/testconsole/testrun/7599824494943412/ for the failures that are now skipped in fbcode

Differential Revision: D41502369

fbshipit-source-id: dd42d54ab993dba53409613251039365901833c9

davidberard98 commented:
@pytorchbot merge
